src/main/scala/is/hail/io/vcf/LoadVCF.scala

```diff
@@ -202,6 +375,51 @@ object LoadVCF {
       .toArray
   }
 
+  // parses the Variant (key), leaves the rest to f
+  def parseLines[C](makeContext: () => C)(f: (C, String, RegionValueBuilder) => Unit)(
+    lines: RDD[WithContext[String]], t: Type): RDD[RegionValue] = {
+    lines.mapPartitions { it =>
+      new Iterator[RegionValue] {
+        val region = MemoryBuffer()
+        val rvb = new RegionValueBuilder(region)
+        val rv = RegionValue(region)
+
+        val context: C = makeContext()
+
+        var present: Boolean = false
+
+        def hasNext: Boolean = {
+          while (!present && it.hasNext) {
+            present = it.next().map { l =>
+              val vcfLine = new VCFLine(l)
+              region.clear()
+              rvb.start(t)
+              rvb.startStruct()
+              val b = vcfLine.parseAddVariant(rvb)
+              if (b) {
+                f(context, l, rvb)
+
+                rvb.endStruct()
+                rv.setOffset(rvb.end())
+              } else
+                rvb.clear()
+              b
+            }.value
```

Comment:
We need to make the WithContext interface a little better. In a future PR we should add `def get[U](f: T => U): U` which would do the same as `map(f).value` without intermediate allocation.

Reply:
`get` taking a function seems strange to me.  What about `mapAndGet`?  It was such a good idea but the code has never seemed quite right to me.  Maybe you need monadic consumers to make it really work out.